geomesa-hbase/geomesa-hbase-rpc/src/main/scala/org/locationtech/geomesa/hbase/rpc/coprocessor/GeoMesaHBaseCallBack.scala

```diff
@@ -8,15 +8,38 @@
 
 package org.locationtech.geomesa.hbase.rpc.coprocessor
 
-import java.util.concurrent.{ConcurrentLinkedQueue, LinkedBlockingQueue}
+import java.util.concurrent.LinkedBlockingQueue
 
 import com.google.protobuf.ByteString
+import com.typesafe.scalalogging.LazyLogging
 import org.apache.hadoop.hbase.client.coprocessor.Batch.Callback
+import org.locationtech.geomesa.hbase.proto.GeoMesaProto.GeoMesaCoprocessorResponse
+import org.locationtech.geomesa.utils.index.ByteArrays
 
-class GeoMesaHBaseCallBack extends Callback[java.util.List[ByteString]] {
+class GeoMesaHBaseCallBack(result: LinkedBlockingQueue[ByteString]) extends Callback[GeoMesaCoprocessorResponse] with LazyLogging {
+  var lastRow: Array[Byte] = _
 
-  val result = new LinkedBlockingQueue[ByteString]()
+  override def update(region: Array[Byte], row: Array[Byte], response: GeoMesaCoprocessorResponse): Unit = {
+    Option(response).map(_.getVersion).foreach { i =>
+      logger.debug(s"Got an empty response.  Coprocessors responded with version $i.")
```

Comment:
should this be a warning/error? we might want to tweak the message - we assume the response is empty but that's just a convention right?

Reply:
Yes, that's just a convention at this point.

Reply:
I'll update this a little bit.